sim/mage/flame_orb.go

```diff
@@ -0,0 +1,160 @@
+package mage
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/stats"
+)
+
+func (mage *Mage) registerFlameOrbSpell() {
+
+	// Frostfire Orb talent converts Flame Orb spell into Frostfire Orb spell.
+	// Don't allow a user access to both at the same time.
+	if mage.Talents.FrostfireOrb != 0 {
+		return
+	}
+
+	mage.FlameOrb = mage.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 82731},
+		SpellSchool:    core.SpellSchoolFire,
+		ProcMask:       core.ProcMaskEmpty, //tbd
+		Flags:          SpellFlagMage | core.SpellFlagAPL,
+		ClassSpellMask: MageSpellFlameOrb,
+
+		ManaCost: core.ManaCostOptions{
+			BaseCost: 0.06,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+			CD: core.Cooldown{
+				Timer:    mage.NewTimer(),
+				Duration: time.Minute,
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, _ *core.Unit, _ *core.Spell) {
+			mage.flameOrb.EnableWithTimeout(sim, mage.flameOrb, time.Second*15)
```

Comment:
I'm not sure if this spell requires to be modeled as a pet. If you really want to be able to show damage from another source unit then sure but it adds some complexities. Right now you're not getting any damage bonuses that the mage would have from temporary buffs and effects for example. Is this expected?

Reply:
The pet version was initially written as such because it executes a rotation where every 1s for 15s it picks a (in practice) random target and hits them. It didn't make the most intuitive sense as a dot since it a) wouldn't make sense to show it as an aura on the enemy and b) would need to be able to randomly do damage to nearby targets. It should be remodeled to be able to benefit from buffs once there's time to do so.

Reply:
If there's a way to hide auras from showing up (i'm sure there is), then there's probably a good way to have an unhasted dot on an enemy that does the spellcast using the random-target-selection I'm currently using.

Reply:
I think you should make it an aura on the mage with an attached PeriodicAction that does your random target spell hit. This would make it nicer on the timeline as well as you can see when you have your flame orb active with other player auras

Reply:
I can take a look at this once this PR is merged and clean it up a bit